docs/DeveloperGuide.md

```diff
@@ -1,38 +1,160 @@
 # Developer Guide
 
+## Table of Contents
+
+<!-- TOC -->
+* [Acknowledgements](#acknowledgements)
+* [Product Scope](#product-scope) 
+  * [Target User Profile](#target-user-profile)
+  * [Value Proposition](#value-proposition)
+* [Design & Implementation](#design--implementation)
+  * [Architecture](#architecture)
+  * [UI component](#ui-component)
+  * [Start component](#start-component)
+  * [Add component](#add-component)
+  * [List component](#list-component)
+  * [View component](#view-component)
+  * [Delete component](#delete-component)
+  * [Exit component](#exit-component)
+* [User Stories](#user-stories)
+  * [V1.0](#v10)
+  * [V2.0](#v20)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Instruction for Manual Testing](#instructions-for-manual-testing)
+<!-- TOC -->
+
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+###### [Back to table of contents](#table-of-contents)
+
+## Product Scope
+
+### Target User Profile
+* People who go to gym and do exercise
+* People who want to lose weight or strength muscles
+
+### Value Proposition
```

Comment:
Perhaps you can further elaborate on how your program features add value to the target user